prov/sockets/src/sock_conn.c

```diff
@@ -481,6 +479,10 @@ struct sock_conn *sock_ep_connect(struct sock_ep *ep, fi_addr_t index)
 out:
 	fastlock_acquire(&ep->cmap.lock);
 	new_conn = sock_conn_map_insert(ep, addr, conn_fd, 0);
+	if (!new_conn) {
+		fastlock_release(&ep->cmap.lock);
+		goto err;
+	}
 	new_conn->av_index = (ep->ep_type == FI_EP_MSG) ? FI_ADDR_NOTAVAIL : (fi_addr_t) idx;
 	conn = idm_lookup(&ep->av_idm, index);
 	if (conn == SOCK_CM_CONN_IN_PROGRESS) {
```

Comment:
The sock_ep_connection function is goto hell and needs to be re-written.  The change itself looks okay.